polaris/ocean/tasks/geostrophic/__init__.py

```diff
@@ -0,0 +1,206 @@
+            component.add_task(Geostrophic(component=component,
+                                           config=config,
+                                           icosahedral=icosahedral,
+                                           include_viz=include_viz))
+
+
+class Geostrophic(Task):
+    """
+    A convergence test for a configuration in geostrophic balance
+
+    Attributes
+    ----------
+    resolutions : list of float
+        A list of mesh resolutions
+
+    icosahedral : bool
+        Whether to use icosahedral, as opposed to less regular, JIGSAW meshes
+
+    include_viz : bool
+        Include VizMap and Viz steps for each resolution
+    """
+    def __init__(self, component, config, icosahedral, include_viz):
+        """
+        Create the convergence test
+
+        Parameters
+        ----------
+        component : polaris.ocean.Ocean
+            The ocean component that this task belongs to
+
+        config : polaris.config.PolarisConfigParser
+            A shared config parser
+
+        icosahedral : bool
+            Whether to use icosahedral, as opposed to less regular, JIGSAW
+            meshes
+
+        include_viz : bool
+            Include VizMap and Viz steps for each resolution
+        """
+        if icosahedral:
+            prefix = 'icos'
+        else:
+            prefix = 'qu'
+
+        subdir = f'spherical/{prefix}/geostrophic'
+        name = f'{prefix}_geostrophic'
+        if include_viz:
+            subdir = f'{subdir}/with_viz'
+            name = f'{name}_with_viz'
+            link = 'geostrophic.cfg'
+        else:
+            # config options live in the task already so no need for a symlink
+            link = None
+        super().__init__(component=component, name=name, subdir=subdir)
+        self.resolutions = list()
+        self.icosahedral = icosahedral
+        self.include_viz = include_viz
+
+        self.set_shared_config(config, link=link)
+
+        self._setup_steps()
+
+    def configure(self):
+        """
+        Set config options for the test case
+        """
+        super().configure()
+
+        # set up the steps again in case a user has provided new resolutions
+        self._setup_steps()
+
+    def _setup_steps(self):
+        """ setup steps given resolutions """
+        icosahedral = self.icosahedral
+        config = self.config
+        config_filename = self.config_filename
+        if icosahedral:
+            prefix = 'icos'
+        else:
+            prefix = 'qu'
+
+        resolutions = config.getlist('spherical_convergence',
+                                     f'{prefix}_resolutions', dtype=float)
+
+        if self.resolutions == resolutions:
+            return
+
+        # start fresh with no steps
+        for step in list(self.steps.values()):
+            self.remove_step(step)
+
+        self.resolutions = resolutions
+
+        component = self.component
+
+        analysis_dependencies: Dict[str, Dict[str, Step]] = (
```

Comment:
```suggestion
        analysis_dependencies: Dict[str, Dict[float, Step]] = (
```

`resolutions` appears to be a list of floats

Reply:
Excellent point.  Does cosine bell have the same problem?

Reply:
Oddly, `cosine_bell` doesn't give an error but `inertial_gravity_wave` did when I implemented it there. However, this fix works for both.